kolibri/plugins/learn/assets/src/views/LibraryPage.vue

```diff
@@ -296,26 +419,39 @@
             getParams.keywords = this.searchTerms.keywords;
           }
           ContentNodeResource.fetchCollection({ getParams }).then(data => {
-            this.results = data.results;
+            this.results = data.results.map(normalizeContentNode);
             this.more = data.more;
             this.labels = data.labels;
             this.searchLoading = false;
           });
         }
+        // else {
+        //   ContentNodeResource.fetchCollection({ getParams }).then(data => {
+        //     this.labels = data.labels;
+        //   });
+        // }
       },
       searchMore() {
         if (this.displayingSearchResults && this.more && !this.moreLoading) {
           this.moreLoading = true;
           ContentNodeResource.fetchCollection({ getParams: this.more }).then(data => {
-            this.results.push(...data.results);
+            this.results.push(...data.results.map(normalizeContentNode));
             this.more = data.more;
             this.labels = data.labels;
             this.moreLoading = false;
           });
         }
       },
-      toggleInfoPanel(content) {
-        this.sidePanelContent = content;
+      removeFilterTag(value) {
+        let query = JSON.parse(JSON.stringify(this.$route.query));
```

Comment:
This works, but seems unnecessary.

Also, you can use `const` here as you are never reassigning `query` to a different value (even though you are modifying the contents of the object that `query` is pointing to).
```suggestion
    removeFilterTag(value, key) {
        const keyObject = this.searchTerms[key];
        delete keyObject[value];
        this.searchTerms = {
            ...this.searchTerms,
            [key]: keyObject,
        };        
```

This way you lean on exactly the same setter/getter code that all the other selection components are using.